rfc/rfc-79/rfc-79.md

```diff
@@ -0,0 +1,140 @@
+  
+
+#### Enabling table service execution and clean to support cancellation and automatic cleanup
+
+The table service flow will be updated to check the /.cancel folder during a pre-commit check before completing the instant. If the instant conflcts with other ongoing writes or is a target of /.cancel, then all data files will be rolled back and the instant will be transitioned to .aborted. These checks will be performed within a transaction, to guard against users cancelling an already-committed instant. In addition, in order to avoid the scenario of writer executing an instant but having its data files being deleted by a concurrent cancellation, the table service writer will perform heartbeating. If an instant has an active heartbeat it can be requested for cancellation (by adding an instant in /.cancel) but it cannot yet be cleaned up and transitioned to .aborted state - this is sufficient for goal (B). 
+
+The below visualization shows the flow for cancellable table service plans (steps that are already in existing table service flow are grey-ed out) 
+
+![cancel table service lifecycle with lock (3)](https://github.com/user-attachments/assets/b882a79d-b582-45f5-874d-57e255d13ff0)
+
+Having this new .hoodie/.cancel folder (in addition to just the .aborted state) is needed not only to allow any writer to forcibily block an instant from being comitted, but also to avoid a table service writer attempt having to unecessairly re-attempt execution of the instant if the previous attempt issues a cancellation (but had failed during cancellation). The below visualized scenario shows how this next attempt will "short circut" in this manner. This scenario also includes an example of concurrent writers to show how transaction and heartbeating in the above proposed flow will allow correct behavior even in the face of concurrent writers attempting to execute and/or cancel the instant.
+
+![cancel flow table serivce](https://github.com/user-attachments/assets/aa8a363c-48f7-4a6c-ae51-cc0c126983f2)
+
+
+Aside from modifications to the table service execution flow, a new pair of cancel APIs request_cancel and execute_cancel will be added for all writers to use. They will allow a writer to request an instant to be cancelled (by adding it to /.cancel) and transition an instant to the terminal .aborted state, respectively.
+
+The new cancel API request_cancel will perform the following steps
+
+1. Start a transaction
+2. Reload active timeline
+3. If instant is already comitted, abort
```

Comment:
can you clarify 3. not sure I understand.

Reply:
Updated phrasing